compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -1043,9 +1056,9 @@ private BLangFunction desugarParticipantFunction(BLangFunction funcNode,
         BLangLambdaFunction trxMainFunc
                 = createLambdaFunction(funcNode.pos, "$anonTrxParticipantFunc$", Collections.emptyList(),
                                        trxReturnNode, trxMainBody);
-        trxMainWrapperFunc.cachedEnv = trxMainFunc.function.clonedEnv;
-        commitFunc.cachedEnv = env.createClone();
-        abortFunc.cachedEnv = env.createClone();
+        trxMainWrapperFunc.capturedClosureEnv = trxMainFunc.function.clonedEnv;
+        commitFunc.capturedClosureEnv = env.createClone();
+        abortFunc.capturedClosureEnv = env.createClone();
```

Comment:
I think we don't need to clone the env here as well. But let's verify that later. Created #21547 to track this.